test/src/userobjects/ElemSideNeighborLayersVarTester.C

```diff
@@ -0,0 +1,109 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org All rights reserved, see COPYRIGHT
+//* for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT Licensed
+//* under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "ElemSideNeighborLayersVarTester.h"
+
+#include "libmesh/parallel.h"
+
+registerMooseObject("MooseTestApp", ElemSideNeighborLayersVarTester);
+
+template <>
+InputParameters
+validParams<ElemSideNeighborLayersVarTester>()
+{
+  InputParameters params = validParams<ElementUserObject>();
+  params.addClassDescription(
+      "Tests whether a coupled variable is correctly ghosted.  This code needs to be cleaned up");
+  params.addRequiredParam<unsigned int>("rank", "The rank for which the ghosted data are recorded");
+
+  params.registerRelationshipManagers("ElementSideNeighborLayers");
```

Comment:
params.registerRelationshipManagers("ElementSideNeighborLayers", "GEOMETRIC ALGEBRAIC");

Reply:
Thanks, i didn't know that i had to use both GEOMETRIC and ALGEBRAIC.

Reply:
Without really diving in, I'm not sure if the suggested syntax above is correct. I intended for the override types to match up with the RMs. You can see that in the GrainTracker example where I have two RMs. In your case you really should only need an ALGEBRAIC RM, no need to have a GEOMETRIC as well. Oh, BTW - these builtin RMs are all ALGEBRAIC by default so your line of could should simplify to this:

```
params.registerRelationshipManagers("ElementSideNeighborLayers");
```

Looping over Evaluable elements really should give you the larger set (beyond just a single layer of elements). It's not very useful if you can't do that.

Reply:
Having "ALGEBRAIC GEOMETRIC" led to exactly the same behaviour as having "".

Looping over Evaluable elements only gives a single layer of elements: you may check this using your "evaluable.i" test (just look at the gold/evaluable_out.e).

Reply:
@WilkAndy If you checkout our next branch, you should see now that the evaluable elements now includes two layers of ghosted elements just like it should

Reply:
Along those lines, if you now remove your `isSemilocal` check and build your map for all evaluable elements, your test will show u0 as being 1 for all elems but the last and u1 as being 1 for all elems but the first. This will be true for both replicated mesh and distributed mesh

![wilkins-test](https://user-images.githubusercontent.com/10248304/55980202-33a3f400-5c51-11e9-9bc8-ad5f89254c5b.png)